### ISSUE TITLE
iOS crash:  malloc: *** error for object 0x100000105480379: pointer being freed was not allocated

### ISSUE THREAD

**Issue by Overtorment** [Sifir-io/react-native-tor#18]
```
SIGABRT
BlueWallet > BlueWallet(5532,0x16cb6f000) malloc: *** error for object 0x100000105480379: pointer being freed was not allocated
```

Steps to reproduce: 

* start the app
* connect to TOR, do something (HTTP request, socket interaction etc)
* lock iphone
* wait 2 minutes
* unlock iphone
* app crashes

stacktrace I got from our sentry:

```
SIGABRT: BlueWallet > BlueWallet(5532,0x16cb6f000) malloc: *** error for object 0x100000105480379: pointer being freed was not allocated

  ?, in core::ptr::drop_in_place
  ?, in tcp_stream_destroy
  File "Tor.swift", line 265, in Tor.startTcpConn
  File "<compiler-generated>", in Tor.startTcpConn
  File "Tor.swift", line 282, in Tor.startTcpConn
...
(15 additional frame(s) were not displayed)
```

![image](https://user-images.githubusercontent.com/1913337/105213526-b8ba7d80-5b46-11eb-8b0e-ee289b0a6bb4.png)

looking at timestamps here, looks like waiting is not necessary, crash happens when we go in background:

![image](https://user-images.githubusercontent.com/1913337/105213714-f3241a80-5b46-11eb-85c4-eb88e16c504a.png)

**Comment by Overtorment** [comment]
![image](https://user-images.githubusercontent.com/1913337/105214145-865d5000-5b47-11eb-9e2d-77aef34b550a.jpeg)

**Comment by gabidi** [comment]
ufff that's nasty. Thanks for reporting @Overtorment  will get on it.

**Comment by gabidi** [comment]
Issue replicated and confirmed.
Seems limited to TCP stream connections.
Patching now
![Uploading Screen Shot 2021-01-21 at 6.51.35 AM.png…]()

**Comment by gabidi** [comment]
patched in https://github.com/Sifir-io/react-native-tor/pull/19 for iOS.
@Overtorment  please give that branch a shot and let me know.

Will check if Android is affected later in the day

**Comment by Overtorment** [comment]
will do, thanks!

**Comment by Overtorment** [comment]
can confirm that the issue is solved. Im able to lock iPhone, unlock, and everything works (and selftest that checks TOR availability passes)

**Comment by gabidi** [comment]
Fixed by #19
